pkg/controller/direct/servicenetworking/utils.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package servicenetworking
+
+import (
+	"encoding/json"
+
+	"k8s.io/klog/v2"
+)
+
+// ReflectClone creates a deep copy of the given object.
+// It is implemented by serializing and unserializing the object to/from json.
+func ReflectClone[T any](obj T) T {
```

Comment:
One general question about ReflectClone: how can we distinguish field unset vs default? I thought we use krm object mostly for that purpose.

Reply:
Looks like we need a design doc for cookies.  Do you want to write it, or should I?